### PR TITLE
Add the metrics config api

### PR DESCRIPTION
**Pull request by amnonh**
This series is based on top of the seastar relabel config API.

The series adds a REST API for the configuration, it allows to get and set it.

The API is registered under the V2 prefix and uses the swagger 2.0 definition.

After this series to get the current relabel-config configuration:

```
    curl -X GET --header 'Accept: application/json' 'http://localhost:10000/v2/metrics-config/'
``` 

A set config example:
```
    curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' -d '[ \
       { \
         "source_labels": [ \
           "__name__" \
         ], \
         "action": "replace", \
         "target_label": "level", \
         "replacement": "1", \
         "regex": "io_que.*" \
       } \
     ]' 'http://localhost:10000/v2/metrics-config/'
```

This is how it looks like in the UI
image

### CODE REVIEW

**Comment by amnonh** [comment]
V2 - A forced push rebase to resolve a conflict

**Comment by denesb** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/538/

Failed because linker crashed with bus error. Re-kicked.

**Comment by amnonh** [comment]
@denesb I've pulled rebased to master and tests are passing.

File: configure.py [hunk 1]
```diff
@@ -1140,6 +1140,7 @@ def find_headers(repodir, excluded_dirs):
        'api/task_manager_test.cc',
        'api/config.cc',
        Json2Code('api/api-doc/config.json'),
+       Json2Code('api/api-doc/metrics.json'),
```
Comment (denesb) [inline configure.py:1143]: @tchaikov FYI

File: api/system.cc [hunk 2]
```diff
@@ -20,8 +28,77 @@ namespace api {
...
+            return smp::invoke_on_all([&relabels, &failed] {
+                return metrics::set_relabel_configs(relabels).then([&failed](const metrics::metric_relabeling_result& result) {
+                    if (result.metrics_relabeled_due_to_collision > 0) {
+                        failed = true;
+                    }
+                    return;
+                });
+            }).then([&failed](){
+                if (failed) {
+                    throw bad_param_exception("conflicts found during relabeling");
```
Comment (denesb) [inline api/system.cc:95]: We should try adding more details here, can be a follow-up.

**Review by denesb — APPROVED** [review]

File: configure.py (hunk 1 above)
Comment (tchaikov) [reply configure.py:1143]: ack. thanks!

Comment (tchaikov) [reply configure.py:1143]: #14753

**Pull request closed without merging**